### ISSUE TITLE
Store `.on(...)` passes outdated state to reducer for `.pending`

### ISSUE THREAD

**Issue by javier-garcia-meteologica** [effector/effector#522]
`Store.on(...)` method is passing an outdated value to the reducer if a `.pending` event is being observed.

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**

The following demo is very simple. The effect `doSomething` does an asynchronous operation while the `store` keeps track of the state (`init`, `pending`, `done` or `failed`).

https://share.effector.dev/QSupDyZX

```js
const doSomething = createEffect(async () => await new Promise(resolve => setTimeout(resolve, 1000)));
  
const store = createStore({ state: 'init' })
 .on(doSomething.pending, (oldState, pending) => (pending ? { state: 'peding' } : oldState)) // Here oldState is completely outdated
 .on(doSomething.doneData, () => ({ state: 'done' }))
 .on(doSomething.fail, () => ({ state: 'failed' }));
  
counter.watch(state => console.log(store));

doSomething()
```

**What is the current behavior:**

The pending `.on(...)` observer is calling the reducer with a outdated state and this makes impossible to track the state of the asynchronous operation.

```js
 {state: "init"}
 {state: "peding"}
 {state: "done"}
 {state: "peding"}
 ```

**What is the expected behavior:**

```js
 {state: "init"}
 {state: "peding"}
 {state: "done"}
 {state: "done"}
 ```

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**

Fails on effector 22.1.0
Worked on effector 21.8.12

**Comment by zerobias** [comment]
Pending is not an event, it is a boolean store, which show whether effect is running, and indeed, it triggered on start of the effect and on end of the effect. Neither `done` received twice nor `done` flipped back to `pending` is expected behavior and effector 22 is forced you to stop abusing that store without understanding it.

In fact, you don’t need it here:
```ts
const fx = createEffect()

type Status = 'init' | 'pending' | 'done' | 'fail'

const $fxStatus = createStore<Status>('init')
  .on(fx, () => 'pending')
  .on(fx.finally, (_, {status}) => status)
```

**Comment by zerobias** [comment]
I've been thinking about this problem for a few days and came to the conclusion that this bug is real. And although your example is still incorrect (duplicate values in updates should be avoided, it is an anti-pattern), but the problem does exist and needs to be solved

**Comment by zerobias** [comment]
Fixed in [effector 22.1.1](https://github.com/effector/effector/releases/tag/effector%4022.1.1)
